test/surface/CMakeLists.txt

```diff
@@ -10,6 +10,10 @@ PCL_SUBSYS_DEPEND(build "${SUBSYS_NAME}" DEPS ${SUBSYS_DEPS} OPT_DEPS ${OPT_DEPS
 
 if (build)
   if (BUILD_io AND BUILD_features)
+    IF(BUILD_surface_afront_debug)
+        add_definitions(-DAFRONTDEBUG)
+    ENDIF(BUILD_surface_afront_debug)
```

Comment:
Keeping in mind my comment of adding definitions, you should not need to add this here as well. It should come exported the moment you link with the `surface` target.